text-generation-inference/server/text_generation_server/model.py

```diff
@@ -99,14 +111,22 @@ def fetch_model(
     logger.warning(f"{model_id} is not a neuron model: it will be exported using cached artifacts.")
     start = time.time()
     logger.info(f"Exporting model to neuron with config {neuron_config}.")
+    log_cache_size()
     start = time.time()
     model = NeuronModelForCausalLM.from_pretrained(model_id, export=True, **export_kwargs)
     end = time.time()
     logger.info(f"Model successfully exported in {end - start:.2f} s.")
     logger.info(f"Saving exported model to local storage under {export_path}.")
+    log_cache_size()
```

Comment:
Will there be anything cached in `HF_HUB_CACHE`? I thought there would only be a cache in `neuron_cache_path`(`/var/tmp/neuron-compile-cache`) after a fresh export.

Reply:
We need some space to expand the checkpoints and store the cached artifacts when we export a hub model.
Whenever we use the `hubclientlibrary` to fetch something (when calling from_pretrained), it ends up under `HF_HUB_CACHE`, so it is supposed to be writeable.
I therefore use the same destination when exporting.
When we deploy our own containers, we mount a writable `/data` volume.
With SageMaker, we only have a writable `/tmp` volume of 30G that can be expanded to 512G.
This is why I added those logs to identifiy issues when exporting the models.